src/Components/CreateImageWizard/formComponents/ReviewStepTables.js

```diff
@@ -10,8 +10,28 @@ import {
   Thead,
   Tr,
 } from '@patternfly/react-table';
+import PropTypes from 'prop-types';
 
 import { UNIT_GIB, UNIT_MIB } from '../../../constants';
+import { useListRepositoriesQuery } from '../../../store/contentSourcesApi';
+
+const RepoName = ({ repoUrl }) => {
+  const { data, isSuccess, isLoading } = useListRepositoriesQuery({
+    url: repoUrl,
+  });
+  if (!isSuccess || isLoading) {
+    return null;
```

Comment:
I'm not really in favor of returning null. React sees null as a valid value and it will try to render it which can cause unexpected behaviour along the way. 

What we could probably do here is remove the `if` conditional, check for success in the second return and based on that return data or spinner?

Maybe something like this:
```javascript
  return (
    <>
      {isSuccess ? <p>{data.data?.[0].name}</p> : <Spinner isSVG size="md" />}
    </>
  );
```

What do you think?

Reply:
The only issue I have with this is that if there is ever an error, we will show an infinite spinner. Of course, that is very unlikely, but generally I would prefer to display the spinner when `isFetching` is true. (RTKQ hooks provide both `isLoading` and `isFetching`, `isLoading` is true when the FIRST request is in flight, `isFetching` is true when ANY request is in flight).

What about something like this? Both specifically for this PR, and as a general convention for us to use moving forward?

```Typescript
return (
    <>
      {isSuccess ? <p>{data.data?.[0].name}</p>} 
      {isFetching ? <Spinner isSVG size="md" />}
      {isError ? <Alert variant="danger" isInline isPlain title="Error loading repository name" />}
    </>
  );
```

Reply:
its great idea, thank you @regexowl and @lucasgarfield